lib/fog/vsphere/models/compute/server.rb

```diff
@@ -319,6 +320,13 @@ def initialize_scsi_controller
             Fog::Compute::Vsphere::SCSIController.new(self.attributes[:scsi_controller])
           end
         end
+
+        def initialize_virtual_machine_summary
+          if attributes[:summary] && attributes[:summary].props
+            self.attributes[:summary] = Fog::Compute::Vsphere::VirtualMachineSummary.new(self.attributes[:summary].props)
```

Comment:
Is this code initializing an association?

If so what do you think of using `fog` DSL for it?

See examples at this [test](https://github.com/fog/fog-core/blob/master/spec/fog_attribute_spec.rb#L52-L57).